java-test-samples/fargate-lambda-stepfunction-async/async-sqs-lambda-stepfunction-dynamodb/README.md

```diff
@@ -0,0 +1,62 @@
+## Async-sqs-lambda-stepfunction-dynamodb
+A Simple DynamoDB & SQS integration with Spring Boot application and deployed on Fargate.
+
+![img_1.png](doc/async-order-processing.png)
+
+### Description:
+The sample Spring boot application written in Java which performs Order Processing operations. AWS serverless services like Amazon API Gateway, ALB, Fargate, DynamoDB, Step Function & SQS being used to host and test the application flow.
+
+### Application
+This project contains a Java17 maven application with Springboot 3.2.2 and AWS Java SDK 2.17.56 dependencies.
+
+### Pre-requisites
+1) SQS Queue with invoke lambda permission
+2) SQS trigger set for Lambda function
+3) Step function with Dynamodb access
+4) DynamoDB tables
+5) Table Attributes
+   1) order_details ref: src/test/resources/order_details.json
+   2) product_details ref: src/test/resources/product_details.json
+
+### Steps:
```

Comment:
Need to understand the rationale of this architecture. Most of the patterns we focus on in the serverless testing samples are at the foundation level, therefore, can be imbibed as a micro-pattern. The entire architecture used in this sample need more thinking or explanation around why API GW with ALB, then Fargate, then SQS, then Lambda function is used. How feasible is this architecture in actual enterprise scenario?

Reply:
We attempted to keep this pattern more in line with the enterprise level ones for asynchronous based scenarios.  

In enterprise scenarios, APIs are typically not exposed directly to external customers, instead, they are exposed through API gateways, which allow for better central authentication and authorization mechanisms, providing static auth tokens, and other example like integration with OKTA and rate limiting. Additionally, API gateways allow for the provision of separate tokens to different customers, allowing for the analysis of how different customers are using the APIs and the transformation of requests/responses by attaching security headers, etc. 
In order to prevent the APIs operating behind the load balancers from being immediately exposed on the internet, the API gateway will make calls to the internal load balancer.

Fargate is used in this pattern to execute containers in a serverless fashion,
We typically utilize Netflix Conductor kind of platforms to orchestrate microservices and business processes, however, as serverless pattern, we attempted to accomplish it using AWS step functions. SQS is used to achieve the asynchronous processing of the requests.

Reply:
I would agree with Dhiraj here. All the patterns we have in this repo can serve as references in bigger application tests. I too think this and also the aync one is little bit an overkill to show the sync and ayns test patterns. I do like the idea of incorporating Fargate in the fix though to give real work scenario, but it can still be simplified.